data_model/src/metadata.rs

```diff
@@ -245,20 +245,20 @@ mod tests {
         let mut metadata = Metadata::new();
         let limits = Limits::new(10, 15);
         metadata
-            .insert_with_limits("0".to_owned(), Metadata::new().into(), limits)
+            .insert_with_limits(Name::test("0"), Metadata::new().into(), limits)
```

Comment:
Why is the function that creates a new name called `test`?

Reply:
Because it instantly constructs `Name` assuming the inputs is valid.
In contrast, `Name::new` gracefully constructs and returns `Result<Self>`

Reply:
Unchecked?

Reply:
`Name::test` includes parsing and unwrapping so if any whitespaces it panics

Reply:
I would recommend here:
1. `new` -> `parse` because this is essentially wrapper for `String::parse`
2. `test` -> `new`/`parse_unwrap`

because `test` is very ambiguous

Reply:
I'd like to use `new` for the default constructor.
`test` might be renamed to `instant` or `succeed`

Reply:
`new` intends the default constructor.
`test` might be renamed to `unchecked`, `instant`, or `succeed`